code/asteroid/asteroid.cpp

```diff
@@ -2195,107 +2273,190 @@ to do with the debris of ships that explode, however these
 
-		Asteroid_impact_explosion_ani = -1;
-		required_string("$Impact Explosion:");
-		stuff_string(impact_ani_file, F_NAME, MAX_FILENAME_LEN);
+		if (optional_string("$Impact Explosion:")) {
+			char impact_ani_file[MAX_FILENAME_LEN];
+			stuff_string(impact_ani_file, F_NAME, MAX_FILENAME_LEN);
 
-		if (VALID_FNAME(impact_ani_file)) {
-			int num_frames;
-			Asteroid_impact_explosion_ani = bm_load_animation(impact_ani_file, &num_frames, nullptr, nullptr, nullptr, true);
+			if (VALID_FNAME(impact_ani_file)) {
+				int num_frames;
+				Asteroid_impact_explosion_ani = bm_load_animation(impact_ani_file, &num_frames, nullptr, nullptr, nullptr, true);
+			}
 		}
 
-		required_string("$Impact Explosion Radius:");
-		stuff_float(&Asteroid_impact_explosion_radius);
+		if (optional_string("$Impact Explosion Radius:")) {
+			stuff_float(&Asteroid_impact_explosion_radius);
+		}
 
 		if (optional_string("$Briefing Icon Closeup Model:")) {
 			stuff_string(Asteroid_icon_closeup_model, F_NAME, NAME_LENGTH);
 		}
-		else {
-			strcpy_s(Asteroid_icon_closeup_model, Asteroid_info[ASTEROID_TYPE_LARGE].pof_files[0]);	// magic file from retail
-		}
 
 		if (optional_string("$Briefing Icon Closeup Position:")) {
 			stuff_vec3d(&Asteroid_icon_closeup_position);
 		}
-		else {
-			vm_vec_make(&Asteroid_icon_closeup_position, 0.0f, 0.0f, -334.0f);  // magic numbers from retail
-		}
 
 		if (optional_string("$Briefing Icon Closeup Zoom:")) {
 			stuff_float(&Asteroid_icon_closeup_zoom);
 		}
-		else {
-			Asteroid_icon_closeup_zoom = 0.5f;	// magic number from retail
-		}
+
 	}
 	catch (const parse::ParseException& e)
 	{
-		mprintf(("TABLES: Unable to parse '%s'!  Error message = %s.\n", "asteroid.tbl", e.what()));
+		mprintf(("TABLES: Unable to parse '%s'!  Error message = %s.\n", filename, e.what()));
 		return;
 	}
 }
 
+// Gets the index of an asteroid by name from Asteroid_info
+static int get_asteroid_index(const char* asteroid_name)
+{
+	for (int i = 0; i < (int)Asteroid_info.size(); i++) {
+		if (!stricmp(asteroid_name, Asteroid_info[i].name)) {
+			return i;
+		}
+	}
+
+	return -1;
+}
+
+static void verify_asteroid_splits() 
+{
+	
+	for (int i = 0; i < (int)Asteroid_info.size(); i++) {
+		SCP_vector<int> splits_to_remove;
+
+		for (int j = 0; j < (int)Asteroid_info[i].split_info.size(); j++) {
+			int asteroid_index = get_asteroid_index(Asteroid_info[i].split_info[j].name);
+			if (asteroid_index >= 0) {
+				Asteroid_info[i].split_info[j].asteroid_type = asteroid_index;
+			} else {
+				Warning(LOCATION,
+					"Unknown asteroid %s defined as split type for asteroid %s, removing!",
+					Asteroid_info[i].split_info[j].name,
+					Asteroid_info[i].name);
+				splits_to_remove.push_back(j); // Remove them all after we're done so we don't mess with the loop
+			}
+		}
+
+		// Remove errored splits
+		for (int j = 0; j < (int)splits_to_remove.size(); j++) {
+			Asteroid_info[i].split_info.erase(Asteroid_info[i].split_info.begin() + splits_to_remove[j]);
```

Comment:
So this is a pretty big problem.  If we remove more than one entry, this is going to accidentally remove the wrong one.

It's saving the indices.  So if you're going to remove index 1 and 2, when you remove index 1, index 2 becomes index 1, and then when you remove index 2, you're removing the original index 3.

If you can guarantee that `splits_to_remove` is in order, then going through them in reverse avoids the problem.  You could also create a new vector of *valid* split_info entries and then replace the old vector with it.

Reply:
Looks good